src/DynamoCore/Models/WorkspaceModel.cs

```diff
@@ -93,6 +95,15 @@ public virtual void OnRequestNodeCentered(object sender, ModelEventArgs e)
                 RequestNodeCentered(this, e);
         }
 
+
+        public event NotifyCollectionChangedEventHandler NodeListChanged;
+
+        public virtual void OnNodeListChanged(object sender, NotifyCollectionChangedEventArgs e)
+        {
```

Comment:
Can we name this `NodesChanged` and `OnNodesChanged`?  This is consistent with the parameter name and is perfectly readable.

Some thoughts on style (take it or leave it):  We've been trying to avoid non-type-safe `EventHandler` objects.  There's a bunch of code taking this approach for no reason other than mimicry.  Why include an object representing the sender when we have generics?  For example, is there any reason `NotifyCollectionChangedEventHandler` couldn't be `Action<WorkspaceModel, NotifyCollectionChangedEventArgs>`?  

But, that still feels rather verbose and painful to me.  

So, consider making these events more granular - i.e. `NodeAdded`, `NodeRemoved`, `NodesCleared`  I find this makes the code a bit more clean, but that's just a matter of opinion.  Then the associated event types are simply `Action<WorkspaceModel, Node>` and `Action<WorkspaceModel, Node>` and `Action<WorkspaceModel>` respectively.  Although it looks like duplicate code, I think it is cleaner and more to the point.